app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,59 @@
+class OrdersController < ApplicationController
+
+  before_action :set_order, only: [:show, :cancel, :paid, :completed]
+
+  def index
+    @orders = current_user.orders
+  end
+
+  def show
+    if current_user.nil? || !current_user.orders.include?(@order) && !current_user.admin?
+      flash[:danger] = "You can only view your own orders. Please log in with the appropritate credentials."
+      redirect_to root_path
+    end
+  end
+
+  def create
+    if session[:user_id].nil?
+      redirect_to login_path
+    else
+      if create_order(session[:cart])
+        session[:cart] = {}
+        flash[:success] = 'Order was successfully placed!'
+        redirect_to orders_path
+      else
+        flash[:danger] = 'Error during order creation, please try again.'
+        redirect_to cart_path
+      end
+    end
+  end
+
+  def cancel
+    @order.update(status: "cancelled")
+    redirect_to admin_dashboard_path
+  end
+
+  def paid
+    @order.update(status: "paid")
+    redirect_to admin_dashboard_path
+  end
+
+  def completed
+    @order.update(status: "completed")
+    redirect_to admin_dashboard_path
+  end
```

Comment:
Since cancel, paid and completed are all doing almost the same thing, you could combine these all into one method called `update_status(status)` and pass in the status to DRY up the code.